src/views/DashboardView/index.vue

```diff
@@ -182,9 +183,19 @@ export default {
         console.log(`rejoining session: ${path}`)
       } else {
         console.log(`session terminated`)
-        this.$router.push('/')
+        //this.$router.push('/')
+        const user = UserService.getUser()
+        if (!user.isVolunteer) {
```

Comment:
Necessary to add two ifs back to back as both these conditions ( for student and volunteer ) have to be checked explicitly.